db/db_utils.go

```diff
@@ -131,13 +148,27 @@ func EnsureIndex(collection string, index mongo.IndexModel) error {
 	return errors.WithStack(err)
 }
 
-// DropIndex takes in a collection and a slice of keys and drops those indexes
-func DropAllIndexes(collection string) error {
+// DropAllIndexes drops all indexes in the specified collections, returning an
+// error immediately if dropping the indexes in any one of them fails.
+func DropAllIndexes(collections ...string) error {
 	env := evergreen.GetEnvironment()
 	ctx, cancel := env.Context()
 	defer cancel()
-	_, err := env.DB().Collection(collection).Indexes().DropAll(ctx)
-	return errors.WithStack(err)
+	for _, coll := range collections {
+		if _, err := env.DB().Collection(coll).Indexes().DropAll(ctx); err != nil {
+			// DropAll errors if the collection does not exist, so make this
+			// idempotent by ignoring the error for the case of a nonexistent
+			// collection.
+			if mongoErr, ok := err.(driver.Error); ok && mongoErr.NamespaceNotFound() {
+				continue
+			}
+			if cmdErr, ok := err.(mongo.CommandError); ok && cmdErr.HasErrorCode(26) {
```

Comment:
what does this one mean?

Reply:
It's mongo driver is pretty opaque about which error is which and when each will be returned. The [docs for driver.Error](https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.7.3/x/mongo/driver#Error) seem to imply it's some catch-all for command execution errors. However, if you read the [docs for mongo.CommandError](https://pkg.go.dev/go.mongodb.org/mongo-driver@v1.7.3/mongo#CommandError), it's _also_ for command execution errors.

To be clear, error-checking for specific cases in MongoDB is basically always hacky like this. Anytime I've seen a piece of code check for specific error codes, it always has to check many cases because the error-handling is not unified very well.